test/util/malicious/app_test.go

```diff
@@ -0,0 +1,67 @@
+package malicious
+
+import (
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/pkg/wrapper"
+	"github.com/celestiaorg/celestia-app/test/util/testfactory"
+	"github.com/celestiaorg/celestia-app/test/util/testnode"
+	"github.com/stretchr/testify/require"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+)
+
+// TestOutOfOrderNMT tests that the malicious NMT implementation is able to
+// generate the same root as the ordered NMT implementation when the leaves are
+// added in the same order, is able to generate roots when leaves are out of
```

Comment:
[optional] grammar improvement

```suggestion
// added in the same order and can generate roots when leaves are out of
```